src/cmd/run.js

```diff
@@ -117,6 +122,18 @@ export default async function run(
   const customPrefs = pref;
   const manifestData = await getValidatedManifest(sourceDir);
 
+  const profileDir = firefoxProfile || chromiumProfile;
+
+  if (profileCreateNew && profileDir) {
+    const isDir = fs.existsSync(profileDir);
```

Comment:
Given that this is already an async function and we are using the fs promise-based API exported by mz, we could easily use the non `*Sync` version of the `fs.exists` and `fs.mkdir` methods and just await on them. 

would you mind to look into making this change?

Reply:
@rpl 
I spent some time looking into making these calls async.
The async version of `fs.existsSync` (`fs.exists`) is deprecated and to make this call `async`, we will have to either use `fs.stat` or `fs.access` inside of a try catch block (to check for the not found exception and only then create the directory) that might make this code a little more verbose than it needs to be.

We could also use the [path-exists library](https://github.com/sindresorhus/path-exists) that might alleviate the need to have a try catch block.

Do you have any other thoughts?

Reply:
@akhilpanchal thanks! that's a very nice research work, I really appreciated that!
That note in the path-exists readme was also a pretty interesting read (the part that warns about the risks of race conditions, in this particular case I don't think we need to worry about that, but it still a good point to keep in mind in general).

Personally I think that we can spare us to add a dependency and just go for `fs.existsSync` for checking if the directory exists and just prefer the async method for the `fs.mkdir` part, does that sound reasonable for you too?

Reply:
Thanks @rpl. Appreciate that :)
Yes, I agree with you that we can spare ourselves from adding a new dependency and just use the async `fs.mkdir` call and keep the sync `fs.existsSync`.
Sending an update to the PR in a few moments.